src/modules/retransmitter/retransmitter.cpp

```diff
@@ -136,33 +148,38 @@ extern "C" void task_retransmitter_fn(void) {
     for (size_t i = 0; i < NUM_MSG; i++) {
         publishers[i]->init(nh);
     }
-    
-    std_msgs::Float64 msg;
-    struct messageAdapter serial;
-    int counter = 0;
-    while(true) {
-        wait_for_msg();
-
-        for (size_t i = 0; i < sizeof(struct message); i++) {
-            serial.data.structBytes[i] = (uint8_t)UARTgetc();
-        }
 
-        /*if (serial.data.msg.endMagic != endMagic) {
-            continue;
-        }*/
+    // while(true) {
+    //     task_read_to_buffer_fn();
+    //     task_publish_buffer_fn();
+    // }
+}

```

Comment:
Tasks should never terminated and add overhead. Can you merge this with one of the other two?

Reply:
Done. We now have two tasks instead of three.